### PR TITLE
gh-101282: move BOLT config after PGO

### PR DESCRIPTION
**Pull request by indygreg**
A future commit will make bolt's configure logic depend on PGO state. I'm moving the block in a standalone commit to make the diff easier to read.

* Issue: gh-101282

### CODE REVIEW

**Comment by indygreg** [comment]
FWIW this change builds up to a build system refactor that fixes some bugs/quirks with BOLT and enables the use of BOLT on libpython, which results in significant pyperformance impact. The remaining rough size of commits are available in #101093.

File: Misc/NEWS.d/next/Build/2023-05-14-19-06-13.gh-issue-104492.7S6WNO.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+BOLT is now configured after PGO. This allows BOLT's configuration to be
```
Comment (corona10) [inline Misc/NEWS.d/next/Build/2023-05-14-19-06-13.gh-issue-104492.7S6WNO.rst]: We don't need NEWS.d with this patch.
No user impact yet all.

**Review by corona10 — APPROVED** [review]
LGTM with nit comment
But for future work, I would like to say.
For the BOLT, it could also be able to be enabled without enabling PGO. (Might be a weird configuration, but someone may need it)

File: Misc/NEWS.d/next/Build/2023-05-14-19-06-13.gh-issue-104492.7S6WNO.rst (hunk 1 above)
Comment (indygreg) [reply Misc/NEWS.d/next/Build/2023-05-14-19-06-13.gh-issue-104492.7S6WNO.rst]: Removed in latest force push.

**Comment by indygreg** [comment]
> LGTM with nit comment But for future work, I would like to say. For the BOLT, it could also be able to be enabled without enabling PGO. (Might be a weird configuration, but someone may need it)

Agree that we should support {none, either, both} configs. But it may require a bit of work to get there. I'll keep this in mind and try to implement it. I think the refactoring I'm doing to the PGO/BOLT make rules should make this vastly easier to achieve.

**Comment by arhadthedev** [comment]
```plain
======================================================================
FAIL: test_find_periodic_pattern (test.test_bytes.ByteArrayAsStringTest.test_find_periodic_pattern) (p='', text='')
Cover the special path for periodic patterns.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/buildbot/bcannon-wasm/3.x.bcannon-wasm.emscripten-node-dl/build/Lib/test/string_tests.py", line 341, in test_find_periodic_pattern
    self.checkequal(reference_find(p, text),
  File "/opt/buildbot/bcannon-wasm/3.x.bcannon-wasm.emscripten-node-dl/build/Lib/test/string_tests.py", line 66, in checkequal
    self.assertEqual(
AssertionError: -1 != 0
```

**Comment by erlend-aasland** [comment]
Post-merge review: LGTM

@indygreg: please don't force-push; it does not play well with reviews on GitHub. If you haven't already, please take a look at the devguide, specifically [Life of a Pull Request](https://devguide.python.org/getting-started/pull-request-lifecycle/)

**Pull request merged**